compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -596,10 +598,22 @@ static LogicalResult lowerWorkgroupCountComputingRegion(
         ArrayRef<OpFoldResult>{workload[tileSize.index()], tileSize.value()});
     workgroupCount.push_back(count);
   }
-  workgroupCount = llvm::to_vector(llvm::reverse(workgroupCount));
   workgroupCount.resize(3, rewriter.getIndexAttr(1));
-  rewriter.replaceOp(workgroupCountOp, getValueOrCreateConstantIndexOp(
-                                           rewriter, loc, workgroupCount));
+  permutedWorkgroupCount.resize(3, rewriter.getIndexAttr(1));
+  int dimId = 0;
+  for (auto map : mapping->getValue()) {
+    auto id = map.cast<mlir::gpu::GPUBlockMappingAttr>().getBlock();
+    if (id == mlir::gpu::Blocks::DimX)
```

Comment:
`permutedWorkgroupCount[static_cast<int64_t>(id)] = workgroupCount[dimId];`

or something equivalent, no need to introduce even more magic constants.